### PR TITLE
SC-4103 - Table-Layout-Improvements

### PR DESCRIPTION
**Pull request by ABialoskorski**
# Issue: [SC-4103](https://ticketsystem.schul-cloud.org/browse/SC-4103)

## Description

I've made changes due to 
image

## Screenshot
image

### CODE REVIEW

**Comment by luizzaly** [comment]
@schul-cloud-bot update screenshots

**Comment by luizzaly** [comment]
could you please upload a screenshot of how the table looks with your changes?

**Comment by ABialoskorski** [comment]
> could you please upload a screenshot of how the table looks with your changes?

sure, done :)

File: src/components/organisms/DataFilter/DataFilterSelect.vue [hunk 1]
```diff
@@ -71,7 +71,7 @@ export default {
 	color: var(--color-tertiary);
 }
 .filter-btn {
-	font-family: var(--font-primary);
+	font-family: var(--font-accent);
```
Comment (adrianjost) [inline src/components/organisms/DataFilter/DataFilterSelect.vue:74]: you do not even need this. For buttons, `-font-accent` is the default font
```suggestion
```

File: src/pages/administration/students/index.vue [hunk 2]
```diff
@@ -30,24 +30,30 @@
...
 						source="material"
 						icon="check"
 						color="var(--color-warning)"
 					/>
 				</span>
-				<span v-else-if="data && data.consentStatus === 'missing'">
+				<span
+					v-else-if="data && data.consentStatus === 'missing'"
+					class="text-content"
+				>
```
Comment (adrianjost) [inline src/pages/administration/students/index.vue:56]: instead of duplicating all the `span` code with these classes I would recommend to use a single wrapping `span` and put in `template`-childs with the according conditions. This would make it easier in the future to update the styles.
```
<span class="text-content">
	<template v-if="condition1">
		content1
	</template>
	<template v-else-if="condition2">
		content2
	</template>
	<!-- ... -->
</span>
```

File: src/pages/administration/teachers/index.vue [hunk 3]
```diff
@@ -32,17 +32,20 @@
...
 						source="material"
 						icon="check"
 						color="var(--color-success)"
 					/>
 				</span>
-				<span v-else-if="data && data.consentStatus === 'missing'">
+				<span
+					v-else-if="data && data.consentStatus === 'missing'"
+					class="text-content"
+				>
```
Comment (adrianjost) [inline src/pages/administration/teachers/index.vue:48]: same applies here

**Comment by adrianjost** [comment]
> * add waiting-for-review label once ready for review


normally mergify should set the "waiting for review" label but it seems to have some problems at the moment. :/

File: src/components/organisms/DataFilter/DataFilterSelect.vue (hunk 1 above)
Comment (ABialoskorski) [reply src/components/organisms/DataFilter/DataFilterSelect.vue:74]: ok, done

Comment (ABialoskorski) [reply src/components/organisms/DataFilter/DataFilterSelect.vue:74]: ok, done ;)

File: src/pages/administration/students/index.vue (hunk 2 above)
Comment (ABialoskorski) [reply src/pages/administration/students/index.vue:56]: (y) done

File: src/pages/administration/teachers/index.vue (hunk 3 above)
Comment (ABialoskorski) [reply src/pages/administration/teachers/index.vue:48]: ok, done

File: src/pages/administration/students/index.vue [hunk 4]
```diff
@@ -30,31 +30,33 @@
...
+						/>
+					</template>
+					<template v-else-if="data && data.consentStatus === 'missing'">
+						<base-icon
+							source="material"
+							icon="close"
+							color="var(--color-danger)"
+						/>
+					</template>
+					<template v-else />
```
Comment (adrianjost) [inline src/pages/administration/students/index.vue:58]: Now that I see this, it can be simplified even more.
1. the template tag can get removed and the condition can be directly at the according base-icon
2. the last v-else without content does nothing, it can be ommited.
```suggestion
				<span class="text-content">
					<base-icon
						v-if="data && data.consentStatus === 'ok'"
						source="custom"
						icon="doublecheck"
						color="var(--color-success)"
					/>
					<base-icon
						v-else-if="data && data.consentStatus === 'parentsAgreed'"
						source="material"
						icon="check"
						color="var(--color-warning)"
					/>
					<base-icon
						v-else-if="data && data.consentStatus === 'missing'"
						source="material"
						icon="close"
						color="var(--color-danger)"
					/>
```

File: src/pages/administration/teachers/index.vue [hunk 5]
```diff
@@ -32,24 +32,26 @@
...
+						/>
+					</template>
+					<template v-else-if="data && data.consentStatus === 'missing'">
+						<base-icon
+							source="material"
+							icon="close"
+							color="var(--color-danger)"
+						/>
+					</template>
+					<template v-else />
```
Comment (adrianjost) [inline src/pages/administration/teachers/index.vue:53]:
```suggestion
					<base-icon
						v-if="data && data.consentStatus === 'ok'"
						source="material"
						icon="check"
						color="var(--color-success)"
					/>
					<base-icon
						v-else-if="data && data.consentStatus === 'missing'"
						source="material"
						icon="close"
						color="var(--color-danger)"
					/>
```

File: src/serverMiddleware/routes.js [hunk 6]
```diff
@@ -7,10 +7,10 @@
...
 module.exports = [
 	`^/administration/datasources?`,
-	// `^/administration/students/?$`,
-	// `^/administration/students/new/?$`,
-	// `^/administration/teachers/?$`,
-	// `^/administration/teachers/new/?$`,
+	`^/administration/students/?$`,
+	`^/administration/students/new/?$`,
+	`^/administration/teachers/?$`,
+	`^/administration/teachers/new/?$`,
```
Comment (adrianjost) [inline src/serverMiddleware/routes.js:13]: should not be part of this PR. We will enable the routes for production here, when all the features are finished. Please do not commit changes in this file till then.
```suggestion
	// `^/administration/students/?$`,
	// `^/administration/students/new/?$`,
	// `^/administration/teachers/?$`,
	// `^/administration/teachers/new/?$`,
```

File: src/pages/administration/students/index.vue (hunk 4 above)
Comment (ABialoskorski) [reply src/pages/administration/students/index.vue:58]: > Now that I see this, it can be simplified even more.
> 
> 1. the template tag can get removed and the condition can be directly at the according base-icon
> 2. the last v-else without content does nothing, it can be ommited.

ok, done :)

**Comment by ABialoskorski** [comment]
> Hi, there are still a couple of things missing :-)
> 
> 1. Make icon of the "Filer hinzufügen" button tertiary (new client)
>    2.The Header Font of the table is medium (as in storybook), the content of the table extra small
> 2. Exchange the sorting icons attached to the ticket, I also attached a link how to do it...
> 
> Thanks

ok, done ;)

**Comment by ABialoskorski** [comment]
> Ah and another thing I just saw- the edit icons must also be tertiary blue.

ok, done ;)

File: src/serverMiddleware/routes.js (hunk 6 above)
Comment (ABialoskorski) [reply src/serverMiddleware/routes.js]: sure ;)

**Comment by adrianjost** [comment]
@schul-cloud-bot update screenshots
